### ISSUE TITLE
[FORMATTING] Spaceship operator (`<=>`) adding unwanted whitespace

### ISSUE THREAD

**Issue by nerixim** [sql-formatter-org/sql-formatter#807]
**Input data**

Which SQL and options did you provide as input?
[LangChain function for vector search.](https://supabase.com/docs/guides/ai/langchain?database-method=sql)

```sql
BEGIN RETURN query
SELECT
  documents.id,
  documents.content,
  documents.metadata AS documents_metadata,
  1 - (documents.embedding <=> query_embedding) AS similarity
FROM
  documents
ORDER BY
  documents.embedding <=> query_embedding
LIMIT
  match_count;

END;
```

**Expected Output**

```sql
BEGIN RETURN query
SELECT
  documents.id,
  documents.content,
  documents.metadata AS documents_metadata,
  1 - (documents.embedding <=> query_embedding) AS similarity
FROM
  documents
ORDER BY
  documents.embedding <=> query_embedding
LIMIT
  match_count;

END;
```

**Actual Output**

```sql
BEGIN RETURN query
SELECT
  documents.id,
  documents.content,
  documents.metadata AS documents_metadata,
  1 - (documents.embedding <= > query_embedding) AS similarity
FROM
  documents
ORDER BY
  documents.embedding <= > query_embedding
LIMIT
  match_count;

END;
```

**Usage**

- How are you calling / using the library?
VSCode (yes, I've set the dialect).
 
- What SQL language(s) does this apply to?
Postgres

- Which SQL Formatter version are you using?
Extension version 1.6.0

**Comment by nene** [comment]
First off, please read the [FAQ](https://github.com/sql-formatter-org/sql-formatter?tab=readme-ov-file#im-having-a-problem-with-prettier-sql-vscode-extension) with regards to this specific VSCode extension.

Second, yep, this operator also looks unsupported in the latest version of the library. Thanks for reporting.

**Comment by nene** [comment]
So, as I understand, this is not a builtin operator in PostgreSQL, but rather a custom one provided by the [pgvector](https://github.com/pgvector/pgvector?tab=readme-ov-file#querying) extension, among several other operators.

So, this really comes down to the formatter not supporting [all possible custom operators](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-OPERATORS) in PostgreSQL.

**Comment by nene** [comment]
Implemented a support for these specific operators. Though the larger problem with custom operators support remains.

**Comment by nene** [comment]
Fixed in 15.4.7
